datafusion/expr/src/utils.rs

```diff
@@ -586,6 +586,12 @@ pub fn from_plan(
             // The preceding part of expr is equi-exprs,
             // and the struct of each equi-expr is like `left-expr = right-expr`.
             let new_on:Vec<(Expr,Expr)> = expr.iter().take(equi_expr_count).map(|equi_expr| {
+                    // SimplifyExpression rule may add alias to the equi_expr.
+                    let equi_expr = match equi_expr {
```

Comment:
Perhaps calling `unalias` might be appropriate

https://github.com/apache/arrow-datafusion/blob/cd4fd80dd5553ffdac41e21c2b7bfe0296f30695/datafusion/optimizer/src/utils.rs#L308

Reply:
We can not call the above `unalias`, because it is in the `optimizer` crate and `from_plan` is in the `expr` crate.

I found another `unalias` in the `expr` crate, and it seems enough to this pr.
https://github.com/apache/arrow-datafusion/blob/41c72cf515389e90c20433dbcc4116a59016a15b/datafusion/expr/src/expr.rs#L711-L717